### PR TITLE
TestImageDistributor: Add setting for forbidden registries

### PR DESCRIPTION
**Pull request by hongkailiu**
Follow up https://github.com/openshift/ci-tools/pull/1165#issue-473338328

/cc @alvaroaleman

### CODE REVIEW

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 1]
```diff
@@ -548,3 +554,16 @@ func upsertObject(ctx context.Context, c ctrlruntimeclient.Client, obj crcontrol
 	}
 	return err
 }
+
+var deniedRegistries = sets.NewString("default-route-openshift-image-registry.apps.build01.ci.devcluster.openshift.com",
```
Comment (alvaroaleman) [inline pkg/controller/test-images-distributor/test_images_distributor.go:558]: This list is guaranteed to expand over time. Can we make this configurable?

Comment (alvaroaleman) [reply pkg/controller/test-images-distributor/test_images_distributor.go:558]: Also, please add a testcase for this

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 2]
```diff
@@ -548,3 +554,17 @@ func upsertObject(ctx context.Context, c ctrlruntimeclient.Client, obj crcontrol
 	}
 	return err
 }
+
+var deniedRegistries = sets.NewString()
```
Comment (petr-muller) [inline pkg/controller/test-images-distributor/test_images_distributor.go:558]: Eh, a global? That's not nice, we should avoid that IMO.

File: pkg/controller/test-images-distributor/test_images_distributor_test.go [hunk 3]
```diff
@@ -376,6 +382,28 @@ func TestReconcile(t *testing.T) {
 				return nil
 			},
 		},
+		{
```
Comment (petr-muller) [inline pkg/controller/test-images-distributor/test_images_distributor_test.go:385]: Looks like we treat pullspecs without `/` in a special way, we should test that behavior too.

Comment (hongkailiu) [reply pkg/controller/test-images-distributor/test_images_distributor_test.go:391]: This is actually a bug. My bad.
After copying from https://github.com/openshift/ci-tools/pull/1067/files#diff-195d249eb017f311ef48ddb48944ae74R385, I forgot the change it to true.

File: pkg/controller/test-images-distributor/test_images_distributor.go (hunk 1 above)
Comment (hongkailiu) [reply pkg/controller/test-images-distributor/test_images_distributor.go]: Done.

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 4]
```diff
@@ -548,3 +557,11 @@ func upsertObject(ctx context.Context, c ctrlruntimeclient.Client, obj crcontrol
 	}
 	return err
 }
+
+func isImportDenied(pullSpec string, deniedRegistries sets.String) bool {
+	i := strings.Index(pullSpec, "/")
+	if i == -1 {
```
Comment (alvaroaleman) [inline pkg/controller/test-images-distributor/test_images_distributor.go:563]: This is wrong and will imports for something like `golang:1.15`

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 5]
```diff
@@ -73,6 +74,7 @@ func AddToManager(mgr manager.Manager,
 		pullSecretGetter:         pullSecretGetter,
 		successfulImportsCounter: successfulImportsCounter,
 		failedImportsCounter:     failedImportsCounter,
+		deniedRegistries:         deniedRegistries,
```
Comment (alvaroaleman) [inline pkg/controller/test-images-distributor/test_images_distributor.go:77]: can we rename this to `forbiddenRegistries` (and everywhere else as well)? `denied` means it was denied in the past, not that it is currently forbidden

File: cmd/dptp-controller-manager/main.go [hunk 6]
```diff
@@ -105,6 +107,7 @@ func newOpts() (*options, error) {
 	flag.Var(&opts.testImagesDistributorOptions.additionalImageStreamTagsRaw, "testImagesDistributorOptions.additional-image-stream-tag", "An imagestreamtag that will be distributed even if no test explicitly references it. It must be in namespace/name:tag format (e.G `ci/clonerefs:latest`). Can be passed multiple times.")
 	flag.Var(&opts.testImagesDistributorOptions.additionalImageStreamsRaw, "testImagesDistributorOptions.additional-image-stream", "An imagestream that will be distributed even if no test explicitly references it. It must be in namespace/name format (e.G `ci/clonerefs`). Can be passed multiple times.")
 	flag.Var(&opts.testImagesDistributorOptions.additionalImageStreamNamespacesRaw, "testImagesDistributorOptions.additional-image-stream-namespace", "A namespace in which imagestreams will be distributed even if no test explicitly references them (e.G `ci`). Can be passed multiple times.")
+	flag.Var(&opts.testImagesDistributorOptions.deniedRegistriesRaw, "testImagesDistributorOptions.denied-registry", "The hostname of an image registry from which there is no synchronization of its images.")
```
Comment (alvaroaleman) [inline cmd/dptp-controller-manager/main.go:110]: Please mention that this can be passed multiple times

**Comment by alvaroaleman** [comment]
/retitle TestImageDistributor: Add setting for forbidden registries

File: pkg/controller/test-images-distributor/test_images_distributor.go (hunk 2 above)
Comment (hongkailiu) [reply pkg/controller/test-images-distributor/test_images_distributor.go]: Done.

File: pkg/controller/test-images-distributor/test_images_distributor.go (hunk 5 above)
Comment (hongkailiu) [reply pkg/controller/test-images-distributor/test_images_distributor.go]: Done.

File: cmd/dptp-controller-manager/main.go (hunk 6 above)
Comment (hongkailiu) [reply cmd/dptp-controller-manager/main.go]: Done.

**Pull request merged**